Triton_Inference_Server_Python_API/README.md

```diff
@@ -0,0 +1,265 @@
+
+The tutorial and Python API package are designed to be installed and
+run within the `nvcr.io/nvidia/tritonserver:23.12-py3` docker image.
+
+A set of convenience scripts are provided to create a docker image
+based on the `nvcr.io/nvidia/tritonserver:23.12-py3` image with the
+Python API installed plus additional dependencies required for the
+examples.
+
+### Trition Inference Server 23.12 + Python API
+
+#### Build Image
+```bash
+   ./build.sh
+```
+
+#### Example Output
+```bash
+#18 naming to docker.io/library/triton-python-api:r23.12 0.0s done
+#18 DONE 0.2s
++ [[ TEST == TRT_LLM ]]
++ [[ TEST == TEST ]]
++ mkdir -p /home/user/tutorials/Triton_Inference_Server_Python_API/models
++ cp -rf /home/user/tutorials/Triton_Inference_Server_Python_API/deps/test/test_api_models/test /home/user/tutorials/Triton_Inference_Server_Python_API/models/.
+```
+
+#### Supported Backends
+```
+dali  fil  identity  onnxruntime  openvino  python  pytorch  repeat  square  tensorflow  tensorrt
+```
+
+#### Included Models
+
+The default build includes a `test` model that can be used for
+exercising basic operations including sending input tensors of
+different data types. The `test` model copies provided inputs of
+`shape [-1, -1]` to outputs of shape `[-1, -1]`. Inputs are named
+`data_type_input` and outputs are named `data_type_output`
+(e.g. `string_input`, `string_output`).
+
+
+## Hello World
+
+### Start Container
+
+The following command starts a container and volume mounts the current
+directory as `workspace`.
+
+```bash
+   ./run.sh
+```
+
+#### Example Output
+
+```bash
+=============================
+== Triton Inference Server ==
+=============================
+
+NVIDIA Release 23.12 (build 77457706)
+Triton Server Version 2.41.0
+
+Copyright (c) 2018-2023, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+
+Various files include modifications (c) NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+
+This container image and its contents are governed by the NVIDIA Deep Learning Container License.
+By pulling and using the container, you accept the terms and conditions of this license:
+https://developer.nvidia.com/ngc/nvidia-deep-learning-container-license
+
+NOTE: CUDA Forward Compatibility mode ENABLED.
+  Using CUDA 12.3 driver version 545.23.08 with kernel driver version 525.85.12.
+  See https://docs.nvidia.com/deploy/cuda-compatibility/ for details.
+
+root@user-machine:/workspace#
+```
+
+### Python Shell
+
+```bash
+python3
+```
+
+#### Example Output
+
+```bash
+Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+>>>
+```
+### Create and Start a Server Instance
+
+```python
+import tritonserver
+
+server = tritonserver.Server(model_repository="/workspace/models")
+server.start()
+```
+
+### List Models
+
+```
+server.models()
+```
+
+#### Example Output
+```python
+{('test', 1): {'name': 'test', 'version': 1, 'state': 'READY'}}
```

Comment:
I know it might be self-explanatory but it might be worth adding some comments about this output (i.e., the key is model name and version. Also, I'm not sure why model name and version is included as both key and value. Perhaps we could include it only in key and remove it from value?